Add mirror support check

linsword13 opened the pull request:
With this change, package managers that don't implement the `mirror_software` phase will error out when execute `workspace mirror`

linsword13 committed:
Add mirror support check

With this change, package managers that don't implement the
`mirror_software` phase will error out when execute `workspace mirror`

linsword13 committed:
Update doc

douglasjacobsen commented on lib/ramble/ramble/application.py:
```diff
@@ -1224,6 +1224,21 @@ def _inputs_and_fetchers(self, workload=None):
+            # from individual package managers.
+            mirror_phases = self.get_pipeline_phases("mirror")
+            if "mirror_software" not in mirror_phases:
+                logger.die(
```
Does this break mirroring inputs only? If so, I think we should not die here, and maybe change to only skip trying to mirror software.

(i.e. it's fine if we only mirror inputs during the mirror pipeline)

linsword13 replied:
Oh very good catch, initially I thought since this check is itself a phase, an invocation of `--phases mirror-inputs` would just work since this check phase doesn't get selected, but then indeed I forgot the `--include-phase-dependencies` option, which would bring in this check phase if specified.

Updated to not error out in such cases. (The code does seem to be somewhat more convoluted now, so definitely open for better ideas.)

douglasjacobsen reviewed (commented):
One question, just to make sure I understand.

linsword13 committed:
Do not fail mirror when phases are explicitly given

Example:

```
ramble workspace mirror -d mypath/ --phases mirror_inputs --include-phase-dependencies
```
